src/libNode/ViewChangeBlockProcessing.cpp

```diff
@@ -238,19 +204,29 @@ bool Node::ProcessVCBlockCore(const VCBlock& vcblock) {
     return false;
   }
 
-  {
-    lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
-    for (unsigned int x = 0; x < newCandidateLeader; x++) {
-      // TODO: If VC select a random
-      // leader, we need to change the way
-      // we update ds composition.
-      UpdateDSCommiteeCompositionAfterVC(*m_mediator.m_DSCommittee);
-    }
-  }
+  UpdateDSCommiteeCompositionAfterVC(vcblock, *m_mediator.m_DSCommittee);
 
   return true;
 }
 
+void Node::UpdateDSCommiteeCompositionAfterVC(
+    const VCBlock& vcblock, deque<pair<PubKey, Peer>>& dsComm) {
+  for (const auto& faultyLeader : vcblock.GetHeader().GetFaultyLeaders()) {
+    deque<pair<PubKey, Peer>>::iterator it =
+        find(dsComm.begin(), dsComm.end(), faultyLeader);
+
+    // Remove faulty leader from the current
+    if (it != dsComm.end()) {
+      dsComm.erase(it);
+    } else {
+      LOG_GENERAL(WARNING, "Cannot find the ds leader to eject");
```

Comment:
If this condition is also impossible to hit, we can change to FATAL now

Reply:
Changed to FATAL